cmd/snap/cmd_quota.go

```diff
@@ -107,13 +107,29 @@ type cmdSetQuota struct {
 	} `positional-args:"yes"`
 }
 
+func parseQuotas(maxMemory string) (*client.QuotaValues, error) {
+	var mem int64
+
+	if maxMemory != "" {
+		value, err := strutil.ParseByteSize(maxMemory)
+		if err != nil {
+			return nil, err
+		}
+		mem = value
+	}
+
+	return &client.QuotaValues{
+		Memory: quantity.Size(mem),
+	}, nil
+}
+
 func (x *cmdSetQuota) Execute(args []string) (err error) {
-	var maxMemory string
-	switch {
-	case x.MemoryMax != "":
-		maxMemory = x.MemoryMax
+	hasQuotaSet := func() bool {
+		return x.MemoryMax != ""
 	}
 
+	quotaProvided := hasQuotaSet()
```

Comment:
Nitpick, `quotaProvided := (x.MemoryMax != "")` would be equally readable, not sure `hasQuotaSet` is needed given that it's used just once.

Reply:
Good point!